src/inference/Inference.jl

```diff
@@ -123,19 +123,26 @@ end
 ######################
 # Default Transition #
 ######################
+# Default
+# Extended in contrib/inference/abstractmcmc.jl
+getstats(t) = nothing
 
-struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
+Base.@kwdef struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
```

Comment:
IMO in most cases you don't want to use `Base.@kwdef`, in particular if it's a non-user facing struct. I think defining the desired constructors is simpler and clearer here (and we don't need an all-keyword argument constructor).
```suggestion
struct Transition{T, F<:AbstractFloat, S<:Union{NamedTuple, Nothing}}
```